core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -669,6 +669,10 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         if (dynamicConfigManager != null)
           CoreUtils.swallow(dynamicConfigManager.shutdown(), this)
 
+        // Close remote log manager before stopping processing requests, to give a change to any
+        // of its underlying clients (especially in the remote metadata manager) to close gracefully.
+        remoteLogManager.foreach(x => CoreUtils.swallow(x.close(), this))
```

Comment:
This is required because we need to close producer and consumer before the dataplane.